Library/Homebrew/diagnostic.rb

```diff
@@ -127,9 +127,9 @@ def please_create_pull_requests(what = "unsupported configuration")
       end
 
       def examine_git_origin(repository_path, desired_origin)
-        return if !Utils::Git.available? || !repository_path.git?
+        return if !Utils::Git.available? || !GitRepository.git?(repository_path)
 
-        current_origin = repository_path.git_origin
+        current_origin = GitRepository.git_origin(repository_path)
```

Comment:
```suggestion
        current_origin = GitRepository(repository_path).git_origin
```

wondering if something like this would be a nicer API to be able to avoid repeatedly passing the `repository_path` every time and instead create a single instance that you can query with multiple methods.